src/main/java/de/zalando/aruha/nakadi/repository/db/SubscriptionDbRepository.java

```diff
@@ -0,0 +1,82 @@
+package de.zalando.aruha.nakadi.repository.db;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.zalando.aruha.nakadi.domain.Subscription;
+import de.zalando.aruha.nakadi.exceptions.DuplicatedSubscriptionException;
+import de.zalando.aruha.nakadi.exceptions.InternalNakadiException;
+import de.zalando.aruha.nakadi.exceptions.NoSuchSubscriptionException;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+
+import java.io.IOException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Set;
+
+import static com.google.common.collect.Sets.newTreeSet;
+
+public class SubscriptionDbRepository extends AbstractDbRepository {
+
+    private final SubscriptionMapper rowMapper = new SubscriptionMapper();
+
+    public SubscriptionDbRepository(final JdbcTemplate jdbcTemplate, final ObjectMapper objectMapper) {
+        super(jdbcTemplate, objectMapper);
+    }
+
+    public void saveSubscription(final Subscription subscription) throws InternalNakadiException,
+            DuplicatedSubscriptionException {
+
+        try {
+            jdbcTemplate.update("INSERT INTO zn_data.subscription (s_id, s_subscription_object) VALUES (?, ?::jsonb)",
+                    subscription.getId(),
+                    jsonMapper.writer().writeValueAsString(subscription));
+        } catch (JsonProcessingException e) {
+            throw new InternalNakadiException("Serialization problem during persistence of event type", e);
+        } catch (DuplicateKeyException e) {
+            throw new DuplicatedSubscriptionException("Subscription with the same key properties already exists", e);
+        }
+    }
+
+    public Subscription getSubscription(final String id) throws NoSuchSubscriptionException {
+        final String sql = "SELECT s_subscription_object FROM zn_data.subscription WHERE s_id = ?";
+        try {
+            return jdbcTemplate.queryForObject(sql, new Object[]{id}, rowMapper);
+        } catch (EmptyResultDataAccessException e) {
+            throw new NoSuchSubscriptionException("Subscription with id \"" + id + "\" does not exist.", e);
+        }
+    }
+
+    public Subscription getSubscription(final String owningApplication, final Set<String> eventTypes,
+                                        final String useCase)
+            throws NoSuchSubscriptionException, InternalNakadiException {
+
+        final String sql = "SELECT s_subscription_object FROM zn_data.subscription " +
+                "WHERE s_subscription_object->>'owning_application' = ? " +
+                "AND replace(s_subscription_object->>'event_types', ' ', '') = ? " +
```

Comment:
The event-types are sorted when you set event-types for a subscription:
https://github.com/zalando/nakadi/blob/ARUHA-202-subscription-persistence/src/main/java/de/zalando/aruha/nakadi/domain/Subscription.java#L51